From: Patrick Brunschwig <patrick@enigmail.net>
Date: Mon, 5 Mar 2018 17:18:29 +0100
Subject: added preference pEpAutoDownload (defaulting to true) to disable
 downloading of pEp in specific situations

(cherry picked from commit 1d32b9b2c291b8570a9985016d445fe16d72ad7b)
---
 package/installPep.jsm        | 25 ++++++++++++++++++++++---
 package/prefs/defaultPrefs.js |  3 +++
 2 files changed, 25 insertions(+), 3 deletions(-)

diff --git a/package/installPep.jsm b/package/installPep.jsm
index 4b6b21a..48dddd1 100644
--- a/package/installPep.jsm
+++ b/package/installPep.jsm
@@ -20,6 +20,7 @@ Cu.import("resource://enigmail/subprocess.jsm"); /*global subprocess: false */
 Cu.import("resource://enigmail/log.jsm"); /*global EnigmailLog: false */
 Cu.import("resource://enigmail/os.jsm"); /*global EnigmailOS: false */
 Cu.import("resource://enigmail/app.jsm"); /*global EnigmailApp: false */
+Cu.import("resource://enigmail/prefs.jsm"); /*global EnigmailPrefs: false */
 Cu.import("resource://gre/modules/PromiseUtils.jsm"); /* global PromiseUtils: false */
 Cu.import("resource://enigmail/files.jsm"); /*global EnigmailFiles: false */
 
@@ -485,13 +486,18 @@ var EnigmailInstallPep = {
    *     progressListener needs to implement the following methods:
    *       void    onError    ({type: errorType, name: errorName})
    *       void    onInstalled ()
+   * @param manualInstall: Boolean: if true, ignore pEpAutoDownload option
    *
-   * @return Installer object
+   * @return Installer object or null (if not installation started)
    */
 
-  startInstaller: function(progressListener) {
+  startInstaller: function(progressListener, manualInstall = false) {
     EnigmailLog.DEBUG("installPep.jsm: startInstaller()\n");
 
+    if (!manualInstall) {
+      if (!EnigmailPrefs.getPref("pEpAutoDownload")) return null;
+    }
+
     if (gInstallInProgress > 0) return null;
 
     gInstallInProgress = 1;
@@ -507,9 +513,22 @@ var EnigmailInstallPep = {
     return i;
   },
 
-  isPepInstallerAvailable: function() {
+  /**
+   * Determine if pEp installer is available online
+   *
+   * @param manualInstall: Boolean: if true, ignore pEpAutoDownload option
+   *
+   * @return true: installer for current platform is online available
+   *         false: otherwise
+   */
+  isPepInstallerAvailable: function(manualInstall = false) {
     EnigmailLog.DEBUG("installPep.jsm: isPepInstallerAvailable()\n");
 
+    if (!manualInstall) {
+      // don't download anything if auto-download is disabled
+      if (!EnigmailPrefs.getPref("pEpAutoDownload")) return false;
+    }
+
     let inspector = Cc["@mozilla.org/jsinspector;1"].createInstance(Ci.nsIJSInspector);
     let urlObj = null;
 
diff --git a/package/prefs/defaultPrefs.js b/package/prefs/defaultPrefs.js
index 111c66b..827ff3d 100755
--- a/package/prefs/defaultPrefs.js
+++ b/package/prefs/defaultPrefs.js
@@ -203,6 +203,9 @@ pref("extensions.enigmail.warnDownloadContactKeys", true);
 // wrap HTML messages before sending inline PGP messages
 pref("extensions.enigmail.wrapHtmlBeforeSend", true);
 
+// automatically download pepmda if it is available (without askin user)
+pref("extensions.enigmail.pEpAutoDownload", true);
+
 // enable encryption/signing of headers like subject, from, to
 // 1: default: ask user at 1st time use / 0: off /  2: on
 pref("extensions.enigmail.protectedHeaders", 1);
